### PR TITLE
Sp 3544

### PR DESCRIPTION
**Pull request by at88mph**

### CODE REVIEW

File: Dockerfile.build [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ARG JAVA_VERSION=11
+
+FROM gradle:7-jdk${JAVA_VERSION} as builder
```
Comment (SharonGoliath) [inline Dockerfile.build:3]: After the `FROM` directive, some Docker build kits (e.g. Ubuntu 22.04) require `ARG JAVA_VERSION`, with no value set, to build the Docker image correctly.

File: Dockerfile.build [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    && apt install -y rsync xz-utils
+
+ADD https://nodejs.org/dist/v18.18.0/node-v18.18.0-linux-x64.tar.xz /tmp/
+
+RUN cd /tmp/ \
+    && tar xvf node-v18.18.0-linux-x64.tar.xz \
+    && rsync -arvc node-v18.18.0-linux-x64/* /usr/local/
+
+FROM gradle:7-jdk${JAVA_VERSION}
+
```
Comment (SharonGoliath) [inline Dockerfile.build:15]: Same comment as above: `ARG JAVA_VERSION` after the `FROM` directive.

File: Dockerfile.build [hunk 3]
```diff
@@ -0,0 +1,16 @@
+ARG JAVA_VERSION=11
```
Comment (SharonGoliath) [inline Dockerfile.build:1]: Is the name `Dockerfile.build` the accepted standard?

**Review by SharonGoliath — APPROVED** [review]

File: Dockerfile.build (hunk 1 above)
Comment (at88mph) [reply Dockerfile.build:3]: Understood, thanks.  If one requires the `JAVA_VERSION` environment variable inside the build, then this is necessary.  I just need it to identify the image to use.

File: Dockerfile.build (hunk 2 above)
Comment (at88mph) [reply Dockerfile.build:15]: Understood, thanks.  If one requires the `JAVA_VERSION` environment variable inside the build, then this is necessary.  I just need it to identify the image to use.

File: Dockerfile.build (hunk 3 above)
Comment (at88mph) [reply Dockerfile.build:1]: No.  This file is intentionally named differently as it's used to create the build environment.  The Science Portal requires Gradle, JDK 11, and NPM, and if you like keeping your desktop clean, then this is used to build an image that can be used to build the Science Portal.

```sh
docker run --rm -ti -v $(pwd):/app -w /app image-from-dockerfile-build gradle -i clean build test
```

**Pull request merged**